test/performance/benchmarks/load-test/main.go

```diff
@@ -183,5 +183,16 @@ func checkSLA(results *vegeta.Metrics) error {
 		return fmt.Errorf("SLA 3 failed. Errors occurred: %d", len(results.Errors))
 	}
 
+	// SLA 4: making sure the defined vegeta total requests is met
+	var expectedRequests uint64
+	for i := 0; i < len(pacers); i++ {
+		expectedRequests = expectedRequests + uint64(pacers[i].Rate(time.Second)*durations[i].Seconds())
+	}
+	if results.Requests == expectedRequests {
+		log.Printf("SLA 4 passed. total requests is %d", results.Requests)
+	} else {
+		return fmt.Errorf("SLA 4 failed. total requests is %d, expected total requests is %d", results.Requests, expectedRequests)
```

Comment:
From the lat run I see: 

```2023/11/02 16:29:17 SLA 4 failed. total requests is 719996, expected total requests is 720000```

Reply:
Is this an issue with the changes here? Or are the changes surfacing the test is underprovisioned (which I think you mentioned earlier)

I'm trying to figure out if we're good to merge this and do the necessary setup changes in a follow up

Reply:
I would argue this way and go ahead with merging this. We had that even on my recent changes. We should look at why we miss the original SLAs in a new issue. Either the SLAs needs updating (as times are different compared to when istio was built-in) or over time we introduces changes that broke the SLAs but we did not notice.

Reply:
I noticed that [too](https://github.com/knative/serving/pull/14662#issuecomment-1834251513) in the [other PR](https://github.com/knative/serving/pull/1466) but when rebased it was back again see [here](https://github.com/knative/serving/pull/14662#issuecomment-1834310209).
Most of the tests are now stable. It could be that it is under provisioned, but it didn't happen frequently. I didn't want to touch the cluster size or anything and we can't really control variance unless we debug it against the cluster (I don't have access to monitor it). 
One question I have for some all tests is whether we care about overhead like image pulling etc.

Reply:
@xiangpingjiang can you rebase this PR to include @skonto changes?